charts/argo-templates/templates/ingest-file.yaml

```diff
@@ -0,0 +1,231 @@
+                - name: profile-id
+                  value: '{{ "{{inputs.parameters.profile-id}}" }}'
+                - name: source-path
+                  value: gs://{{ "{{inputs.parameters.gcs-bucket}}" }}/{{ "{{inputs.parameters.gcs-file-path}}" }}
+                - name: target-path
+                  value: '{{ "{{inputs.parameters.target-path}}" }}'
+                - name: sa-secret
+                  value: '{{ "{{inputs.parameters.sa-secret}}" }}'
+                - name: sa-secret-key
+                  value: '{{ "{{inputs.parameters.sa-secret-key}}" }}'
+          # 3) Poll the ingest job for the file repeatedly until the file has been ingested
+          - name: poll-ingest-job
+            dependencies: [submit-file-ingest]
+            template: poll-ingest-job
+            when: '{{ "{{tasks.check-existing-file.outputs.result}} == false" }}'
+            arguments:
+              parameters:
+                - name: api-url
+                  value: '{{ "{{inputs.parameters.url}}" }}'
+                - name: job-id
+                  value: '{{ "{{tasks.submit-file-ingest.outputs.result}}" }}'
+                - name: timeout
+                  value: '{{ "{{inputs.parameters.timeout}}" }}'
+                - name: sa-secret
+                  value: '{{ "{{inputs.parameters.sa-secret}}" }}'
+                - name: sa-secret-key
+                  value: '{{ "{{inputs.parameters.sa-secret-key}}" }}'
+          # 4) Return the fileId
+          - name: get-job-result
+            dependencies: [poll-ingest-job]
+            template: get-job-result
+            when: '{{ "{{tasks.check-existing-file.outputs.result}} == false" }}'
+            arguments:
+              parameters:
+                - name: api-url
+                  value: '{{ "{{inputs.parameters.url}}" }}'
+                - name: job-id
+                  value: '{{ "{{tasks.submit-file-ingest.outputs.result}}" }}'
+                - name: sa-secret
+                  value: '{{ "{{inputs.parameters.sa-secret}}" }}'
+                - name: sa-secret-key
+                  value: '{{ "{{inputs.parameters.sa-secret-key}}" }}'
+      outputs:
+        parameters:
+          - name: file-exists
+            valueFrom:
+              parameter: '{{ "{{tasks.check-existing-file.outputs.result}}" }}'
+          - name: file-id
+            valueFrom:
+              parameter: '{{ "{{tasks.get-job-result.outputs.result}}" }}'
+              default: 'null'
+    {{- $secretVolume := "sa-secret-volume" }}
+    {{- $mountPath := "/secret" }}
+    # SubTemplates
+    - name: check-existing-file
+      inputs:
+        parameters:
+          - name: api-url
+          - name: dataset-id
+          - name: target-path
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```

Comment:
Should change to use the new Argo input (or else we need a new value for this Helm chart)

Reply:
It looks like one of the existing input parameters (`sa-secret`) actually has this value, so I switched to using that.